tests/functional/RemoteTargetLocatorTest.php

```diff
@@ -71,4 +72,86 @@ public function testActiveElement()
         $this->assertSame('input', $activeElement->getTagName());
         $this->assertSame('test_name', $activeElement->getAttribute('name'));
     }
+
+    /**
+     * Data provider for frame().
+     */
+    public function frameProvider()
+    {
+        $parentFrame = 'This is the host page which contains an iFrame';
+        $childFrame = 'This is the content of the iFrame';
+        $otherChildFrame = 'open new window';
+
+        return [
+            'null' => [
+                null,
+                $parentFrame,
+            ],
+            'string 0' => [
+                '0',
+                $childFrame,
+            ],
+            'int 0' => [
+                0,
+                $childFrame,
+            ],
+            'float 0' => [
+                0.0,
+                $childFrame,
+            ],
+            'string 1' => [
+                '1',
+                $otherChildFrame,
+            ],
+            'int 1' => [
+                1,
+                $otherChildFrame,
+            ],
+            'floa 1' => [
+                1.0,
+                $otherChildFrame,
+            ],
+        ];
+    }
+
+    /**
+     * @cover ::frame
+     * @dataProvider frameProvider
+     */
+    public function testValidFrames($frame, $content)
+    {
+        $this->driver->get($this->getTestPageUrl('page_with_frame.html'));
+        $this->driver->switchTo()->frame($frame);
+        $this->assertContains(
+            $content,
+            $this->driver->switchTo()->activeElement()->getText()
+        );
```

Comment:
I suggest separating every part of the arrange/act/assert phase of the test by new line. In this case it means newline after eachline:
```
        $this->driver->get($this->getTestPageUrl('page_with_frame.html'));

        $this->driver->switchTo()->frame($frame);

        $this->assertContains(
            $content,
            $this->driver->switchTo()->activeElement()->getText()
        );
```